README.md

```diff
@@ -42,6 +42,33 @@ The services of a client divide the API into logical chunks and correspond to
 the structure of the GitHub API documentation at
 https://developer.github.com/v3/.
 
+
+A basic unauthenticated program might look something like:
+
+```go
+import (
+	"context"
+	"github.com/google/go-github/github"
+)
+
+func FetchOrganizations(username string) ([]*Authorization, error) {
+	ctx := context.Background()
+	client := github.NewClient(nil)
+
+	// list all organizations for user "willnorris"
+	orgs, _, err := client.Organizations.List(ctx, "willnorris", nil)
+
+	if err != nil {
+		return nil, err
+	}
+	return orgs, err
+}
+```
```

Comment:
This program, contrary to what the objective of #809 was, is not actually complete. It's missing a `package main` clause, and it's missing `func main() { ... }`. I think we should fix that and make it a complete, valid program.

However, I think the README is not the best place for it. It becomes a long snippet and feels out of place, given the above paragraphs already cover this.

Instead, I really like the idea of linking to the example directory. So let's make this a complete basic program and move it to the `example` directory.

Reply:
In this example, I attempted to create a simple generic function which can be placed directly into any package. Hence, I thought of excluding the `package` name and `func main` as I wanted to separate the caller from the functionality.

I agree with you that adding a complete program might render the README look clumsy and dense. However, there are two issues that we need to address before we migrate this code to the `example` directory:

1. Adding information regarding the need to import `context` package and defining `ctx` variable in the snippets preceding this example code.
2. Move the example to a `noauth` subdirectory (maybe?) under the `example` directory. 

If we can agree to these suggestions, I can make the required modifications.

P.S: I notice two bugs in the example snippet:
1. use `username` instead of the hardcoded string
2. `[]*Organization` should be the type of the first return value.

Reply:
> Adding information regarding the need to import context package and defining ctx variable in the snippets preceding this example code.

I think this can be resolved by having a comment in the program. It can be a few lines long and explain that [`context`](https://godoc.org/context) package is being used. If there's no context available, then `context.Background()` can be used as a starting point. But it should point reader to the documentation of [`context`](https://godoc.org/context) package for more info.

> Move the example to a noauth subdirectory (maybe?) under the example directory.

I think we should have all examples under `example` directory, let's just give them descriptive names.

I would suggest we call the example you're suggesting `simple` or `hello` or something like that.

We should also link to it right after the `// list all organizations for user "willnorris"` snippet in README.

I'm not completely convinced that having a general function would be helpful in this context. General functions make sense when used multiple times, otherwise they add unnecessary complexity. I'd imagine a simpler example of a complete program would be a better starting point. For example, this is how I imagine `simple` can look like:

```Go
// TODO: Explain what this command does, etc.
package main

import (
	"context"
	"fmt"
	"log"

	"github.com/google/go-github/github"
)

func main() {
	// Create a GitHub API client without authentication.
	client := github.NewClient(nil)

	// List all organizations for user "willnorris".
	orgs, _, err := client.Organizations.List(context.Background(), "willnorris", nil)
	if err != nil {
		log.Fatalln(err)
	}

	// Print organization names.
	for _, org := range orgs {
		fmt.Println(org.Name)
	}
}
```